ngraph/core/reference/include/ngraph/runtime/reference/interpolate.hpp

```diff
@@ -454,62 +450,155 @@ namespace ngraph
+                            for (int64_t i = 0; i < points_in_neighbor; ++i)
+                            {
+                                int64_t offset = 0;
+                                for (int64_t j = 0; j < spatial_rank; ++j)
+                                {
+                                    if (i & (1 << (spatial_rank - 1 - j)))
+                                    {
+                                        offset += in1[j] * input_index_multipliers[j];
+                                    }
+                                    else
+                                    {
+                                        offset += in2[j] * input_index_multipliers[j];
+                                    }
+                                }
+                                values_of_input_points[i] = xdata[offset];
+                            }
+
+                            // 4. Interpolation.
+                            float sum = 0.0f;
+                            for (int64_t i = 0; i < points_in_neighbor; ++i)
                             {
-                                T x11 = xdata[info.input_width_mul_y1[y] + info.in_x1[x]];
-                                T x21 = xdata[info.input_width_mul_y1[y] + info.in_x2[x]];
-                                T x12 = xdata[info.input_width_mul_y2[y] + info.in_x1[x]];
-                                T x22 = xdata[info.input_width_mul_y2[y] + info.in_x2[x]];
-
-                                ydata[output_width * y + x] =
-                                    static_cast<T>(info.dx2[x] * info.dy2[y] * x11 +
-                                                   info.dx1[x] * info.dy2[y] * x21 +
-                                                   info.dx2[x] * info.dy1[y] * x12 +
-                                                   info.dx1[x] * info.dy1[y] * x22);
+                                float coeff = 1.0f;
+                                for (int64_t j = 0; j < spatial_rank; ++j)
+                                {
+                                    coeff *= (i & (1 << (spatial_rank - 1 - j))) ? d1[j] : d2[j];
+                                }
+                                sum += coeff * values_of_input_points[points_in_neighbor - 1 - i];
                             }
+
+                            // 6. Store result.
+                            ydata[idx] = static_cast<T>(sum);
                         }
-                        xdata += input_height * input_width;
-                        ydata += output_width * output_height;
+
+                        xdata += input_data_ptr_increment;
+                        ydata += output_data_ptr_increment;
                     }
                 }
             }
 
+            template <typename T>
+            void InterpolateEval<T>::linear_onnx_func(const T* input_data, T* out)
+            {
+                linear_onnx_generic_func(input_data, out);
```

Comment:
Why do we need to have this one line function?

Reply:
Now this one line function was deleted. But the function `linear_onnx_generic_func` was renamed as `linear_onnx_func`.